### PR TITLE
Make PyCDS schema name agnostic

### LINKED ISSUE #32: Make PyCDS schema (name) agnostic
I recall having a lot of trouble with working within the correct schema (in the sense of PostgreSQL: a collection of tables, etc. within a database) when working on PyCDS 2+ years ago. I suspect it was just ignorance about how to use `search_path`. There is some particularly gorky-looking code in `scripts/manage-views.py` relating to `search_path`. This all needs to be renovated to be schema-agnostic.

### PR DESCRIPTION
**Pull request by rod-glover**
Resolves #32 

This PR replaces the hard-coded schema name `'crmp'` throughout PyCDS with a schema name determined by the value of the environment variable `PYCDS_SCHEMA_NAME`. If `PYCDS_SCHEMA_NAME` is not defined, the schema name defaults to `'crmp'`, making this change fully backward compatible with existing code.

To do:

- [x] Invoke tests with PYCDS_SCHEMA_NAME` defined
- [x] Update README

### CODE REVIEW

**Comment by rod-glover** [comment]
I've addressed both comments (and f-strings!) in the code. Good suggestions.

File: pycds/functions.py [hunk 1]
```diff
@@ -269,47 +278,50 @@
...
 # NOTE: Production code: This function is called by the PDP PCDS backend.
-query_one_station = DDL('''
-    CREATE OR REPLACE FUNCTION query_one_station(station_id integer)
+def query_one_station(schema=get_schema_name()):
+    return DDL(f'''
+    CREATE OR REPLACE FUNCTION {schema}.query_one_station(station_id integer)
       RETURNS text AS
     $BODY$
-        stn_query = "SELECT * FROM crmp.getStationVariableTable(" + str(station_id) + ", false)"
+        stn_query = "SELECT * FROM getStationVariableTable(" + str(station_id) + ", false)"
```
Comment (rod-glover) [inline pycds/functions.py]: `{schema}.getStationVariableTable`

File: pycds/functions.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
 """
 
 from sqlalchemy.schema import DDL
+from pycds import get_schema_name
```
Comment (rod-glover) [inline pycds/functions.py]: There is a set of bugs, or at least shortcomings, introduced in these function definitions. The assumption we are now (mostly) making is that the search path is not to be relied on, and schema name is explicit everywhere. Except that is not the case here. These definitions were copied direct from the CRMP database definitions and they didn't include explicit schema names.

A couple of examples noted in comments below.

**Comment by rod-glover** [comment]
@jameshiebert , regarding the lack of explicit schema name in function bodies pointed out above: I think I should interpolate schema names throughout. It's not hard to do, although without tests for every function, it would be easy to make an undetected mistake. Your thoughts?

**Comment by jameshiebert** [comment]
> I should interpolate schema names throughout.

Sounds good to me.

>It's not hard to do, although without tests for every function, it would be easy to make an undetected mistake. Your thoughts?

Yes that's true, though with a careful read, I'd think you *should* be able to find them all.

**Comment by rod-glover** [comment]
@jameshiebert , I had a little feeling that not all was well in all functions. The smoke tests proved out that intuition, and I corrected most of the problems. Definitely worth the effort.

Two problems remain: `daily_ts` and `monthly_ts` have errors in their definitions in the CRMP database and those errors were transcribed to this package. I will put in an issue to fix them, but that will require intervention by Faron, I think. I'll @ mention him in the issue. Meanwhile the failing tests are marked `xfail`.

Any further comments?

**Comment by jameshiebert** [comment]
No further comments from me.

**Pull request merged**